### PR TITLE
Add next alarm sensor, and sort alarms and timers

### PR DESCRIPTION
**Pull request by DurgNomis-drol**
As described. This sorts alarms and timers and adds a new sensor called sensor.*_next_alarm which only shows the next alarm and it's attributes.

### CODE REVIEW

File: custom_components/ha-google-home/sensor.py [hunk 1]
```diff
@@ -88,6 +94,51 @@ def _get_alarms_data(self):
...
+        super().__init__(coordinator, entry)
+        self._name = device_name
+
+    @property
+    def state(self):
+        alarms = self._get_alarm_data()
+        state = alarms[0][LOCAL_TIME_ISO] if len(alarms) else STATE_OFF
+        # The first one will always be the closet one
+        # as we have sorted the list in _get_alarm_data()
+        return state
```
Comment (KapJI) [inline custom_components/ha-google-home/sensor.py:115]: Multiple changes here:
```suggestion
        # The first one will always be the closest one
        # as we have sorted the list in _get_alarm_data()
        return alarms[0][LOCAL_TIME_ISO] if alarms else STATE_OFF
```

File: custom_components/ha-google-home/entity.py [hunk 2]
```diff
@@ -87,6 +88,41 @@ def state(self):
...
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return ICON_ALARMS
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
```
Comment (KapJI) [inline custom_components/ha-google-home/entity.py:123]: This is already defined in `GoogleHomeNextAlarmSensor`
```suggestion
```

**Review by KapJI — CHANGES_REQUESTED** [review]

File: custom_components/ha-google-home/entity.py [hunk 3]
```diff
@@ -87,6 +88,37 @@ def state(self):
...
+        """Return a unique ID to use for this entity."""
+        return f"{self._name}/{LABEL_NEXT_ALARM}"
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return ICON_ALARMS
+
+
+
```
Comment (KapJI) [inline custom_components/ha-google-home/entity.py:121]: Now there're too many blank lines 🙂

**Review by KapJI — APPROVED** [review]
Looks good to me!

**Comment by leikoilja** [comment]
This is sooo cool, thanks guys for picking it up! 👍 
@DurgNomis-drol have you tried the sensor with multiple alarms per device where some are recurring ones? I'm wondering if `fire_time` is dynamically changed to which alarm is the soonest on google device that we can rely on?

Also, now that we are creating next_alarm, should we maybe consider making next_timer as well? We can reuse most of logic, but it would be consistent and probably useful as well?

**Comment by DurgNomis-drol** [comment]
Thanks @leikoilja 😄 

I believe it does change fire_time dynamically, but it is something i we need to confirm before merging

It would be very simple to add a next timer sensor as well, I will add it as soon as you have rebased it 😄

**Comment by leikoilja** [comment]
@DurgNomis-drol, you can try adding the next timer sensor now :) I'll test a little bit in the meanwhile.

**Comment by DurgNomis-drol** [comment]
It does dynamically update `fire_time` in about 1 min or so.

**Comment by leikoilja** [comment]
That's exactly what we need! 🚀  I have also just been testing and it seems to work really good! 💥

**Comment by leikoilja** [comment]
@DurgNomis-drol do you mind updating the `README` to add there information about next alarm/timer as well? :)

**Comment by DurgNomis-drol** [comment]
@leikoilja Good to here! 🚀 

And yes, will do it as the last thing before we merge this one 😄

**Review by leikoilja — CHANGES_REQUESTED** [review]
Will request changes just to keep in loop :) Please re-request when we are ready to merge 💥

**Comment by DurgNomis-drol** [comment]
And wow the integration got fast with the latest PR 😄 This is so awesome!

File: custom_components/ha-google-home/sensor.py [hunk 4]
```diff
@@ -32,6 +36,12 @@ async def async_setup_entry(hass, entry, async_add_devices):
                             device.device_name,
                             getattr(device, LABEL_ALARMS),
                         ),
+                        GoogleHomeNextAlarmSensor(
+                            coordinator,
+                            entry,
+                            device.device_name,
+                            getattr(device, LABEL_ALARMS),
+                        ),
```
Comment (leikoilja) [inline custom_components/ha-google-home/sensor.py:44]: You are forgetting to initialize the next timer sensor here :)

**Review by leikoilja — APPROVED** [review]
Awesome stuff! Thanks for implementing it! 💥 

Please see my small comment. 
Also please **test that it works** and if so - let's merge it in 🚀 
It's a good habit to test something you have implementing before merging. Some of us that review can only spot some technicalities without testing if it works as excepted. And once it's merged it can potentially break `master` if not tested well(obv not in this particular implementation). And breaking `master` would not be good 😅 
Thanks again 💯

**Comment by DurgNomis-drol** [comment]
I don't know why it disappeared, I added it, because i tested it earlier.

I always test, even before committing to my own branch. 😄  But i haven recently moved to my Windows 10 Desktop from a Linux laptop and Windows does not play well with pre-commit.

Sorry for any problems or confusion it have made. I will try to be better at spotting if something happens that should not happen in the future. 😄

**Comment by leikoilja** [comment]
Oh, no worries at all! You are doing an awesome work 💥 keep it up 🚀

**Pull request merged**